File: LeetCode/3110. Score of a String/3110.java
class Solution {
    public int scoreOfString(String s) {
        // Initialize a variable to keep track of the score. The initial value is 0.
        int score = 0;

        // Loop through all characters of the string. We start from the second character because we will compare it with the previous character.
        for (int i = 1; i < s.length(); i++) {
            // Subtract the ASCII value of the previous character from the ASCII value of the current character.
            // We take the absolute value of this subtraction to avoid negative results.
            // We add this value to the total score.
            score += Math.abs(s.charAt(i) - s.charAt(i - 1));
        }

        // Return the total score.
        return score;
    }
}

/*
 class Solution {
    public int scoreOfString(String s) {
        return iterateString(0,s);   
    }

    public static int iterateString(int n, String s){
        if(n == s.length() -1 )
            return 0;
        return Math.abs(s.charAt(n) - s.charAt(n+1)) + iterateString(n+1, s);
    }
}
 */

File: LeetCode/3110. Score of a String/3110.py
class Solution(object):
    def scoreOfString(self, s):
        """
        :type s: str
        :rtype: int
        """
        score = 0  # Initialize a variable to keep track of the score. The initial value is 0.
        for i in range(1, len(s)):  # Loop through all characters of the string, starting from the second character
            score += abs(ord(s[i]) - ord(s[i - 1]))  # Subtract the ASCII value of the previous character from the ASCII value of the current character, take the absolute value of this subtraction, and add it to the total score.
        return score  # Return the total score.

'''
class Solution(object):
    def scoreOfString(self, s):
        """
        :type s: str
        :rtype: int
        """
        x = 0
        for i in range(1,len(s)):
            x+= abs(ord(s[i-1])-ord(s[i]))
        return x
'''

'''
class Solution(object):
    def scoreOfString(self, s):
        """
        :type s: str
        :rtype: int
        """
        temp = ord(s[0])
        count = 0
        for i in s:
            a = ord(i)
            count += abs(temp-a)
            temp = a
        return count
'''